Preparing for new kinds of monitoring

gleopoldo opened the pull request:
I'm aiming towards solving both #9 and #10. IMHO, there're some changes that were required so that we can extend this project with new features and monitorings. So, the main changes are:

* Added a dev environment with docker. This looks much easier, IMHO, to work on a project, since anyone wouldn't need to handle different versions of ruby, it's dependencies and so on, letting everything else to an automated script that sets up his or her dev environment;

* Refactored the current checks to use external classes instead, this way we can group related monitorings into a single place, for example: a Redis class is responsible for monitoring redis stuff, while a Sidekiq class monitors sidekiq stuffs.

* Removed the custom error feature: I think that this decision is far more polemic than the other changes, and thus I want to hear your opinions on this... I opted to remove it since I'm going to add new monitoring in another PR that breaks with how the `custom_error`  method works because it's template maybe won't apply into new error messages. I thought to use a lambda, or a proc, and pass down the `custom_error` to translate specific errors, but this can be overkill. WDYT?

Commit:
Dockerize dev environment

Commit:
Refactor redis validations

Commit:
Removed custom error messages

Since we're going to implement a process monitoring, the custom error
message feature should be adapted to the new possible sets of errors
that may arrise with this kind of monitoring. Having a feature to
control just parts of possible errors seems overkill in this scenario

Commit:
Add a CLI to setup dev environment

Commit:
Solve rubocop issues

Commit:
Avoid using heartcheck with incompatible sidekiq version

Commit:
Add restrictions to sidekiq on travis runnning with ruby 2.2.2

fabioperrella commented on docker-compose.yml:
```diff
@@ -0,0 +1,17 @@
+      BUNDLE_PATH: /app/.gems
+
+  redis:
+    image: redis:latest
```
What about setting a specific Redis version and not use `latest` ?

This way helps to identify what is the supported version of Redis and it will not break when there is a major upgrade in Redis.

gleopoldo replied:
Sure! I'm going to lock it in 5.0.6 - which seems to be the latest one on dockerhub, okay?

fabioperrella replied:
good!

fabioperrella commented on Gemfile.lock:
```diff
@@ -1,26 +1,25 @@
   remote: .
   specs:
-    heartcheck-sidekiq (0.1.1)
+    heartcheck-sidekiq (0.1.2)
```
What about removing the Gemfile.lock from git?

For me, this is a [good practice in gem projects](https://yehudakatz.com/2010/12/16/clarifying-the-roles-of-the-gemspec-and-gemfile/)

gleopoldo replied:
Nice one! Removed!

fabioperrella commented on scripts/heartcheck:
```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
```
what about using `bash -e` to stop the script if some error happens?

gleopoldo replied:
I was using a `set -e`, but setting this at the file's head is much cleaner. Tks!

fabioperrella replied:
humm.. great!

fabioperrella commented:
Maybe the next time, you could split this PR in 3, like the items you described! I think this would help to review.

Commit:
Removing Gemfile.lock from git tracking

Locking development environment version of gems dependencies is
not a good practice.

Commit:
Ensure CLi script fails fast
